### ISSUE TITLE
[MFLES] Seasonality prediction incorrect when forecasting multi-seasonal time series beyond one period

### ISSUE THREAD

**Issue by tehunter** [Nixtla/statsforecast#938]
### What happened + What you expected to happen

If you have a time series with multiple seasonality periods, MFLES currently constructs the combined seasonality pattern in an array sized to the largest of the seasonality periods. This leads to an incomplete combined pattern being stored (unless the largest seasonality period happens to be a multiple of all the smaller periods).

For example, in the following example, the data has seasonal periods of 2 and 5. 10 days are needed to correctly capture the combined seasonality, but the forecast uses only a 5-day pattern.

![image](https://github.com/user-attachments/assets/53adb282-a776-4fa5-9769-78e74f98a908)

I believe it's inconsequential if the forecast horizon is less than a full seasonality period.

### Versions / Dependencies

Bug exists in main branch

### Reproducible example

```python
from statsforecast import StatsForecast
from statsforecast.models import (
    MFLES
)
import pandas as pd
import numpy as np

season_1 = np.array([0, 100])
season_2 = np.array([0, -200, 0, 0, 0])
dates = pd.date_range("2024-01-01", "2024-01-30")
n = len(dates)

sample_ts = pd.DataFrame({
    "ds": dates,
    "y": np.resize(season_1, n) + np.resize(season_2, n),
    "unique_id": "1"
})

model = MFLES(
    season_length=[2, 5],
    trend_lr=0,
    residuals_lr=0,
)
sf = StatsForecast(
    models=[model],
    freq="D"
)
sf.fit(
    sample_ts
)

forecast_df = sf.predict(h=30)
sf.plot(sample_ts, forecast_df)
```

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

**Comment by janrth-wlt** [comment]
@tblume1992 are you aware of this?

**Comment by tblume1992** [comment]
> [@tblume1992](https://github.com/tblume1992) are you aware of this?

Yes! @jmoralez and I awhile back took a shot but weren't satisfied with the quick fix. Essentially the seasonality component is held in an array with the length of the max of the season_length list provided. So in the case where the larger seasonality is not a multiple of the smaller you get this issue where it won't line up.
